S4 game progress

SamBellDev opened the pull request:
New functionality in next question and reveal answer button.

SamBellDev committed:
Created reveal button functionality

MrLukey committed:
Merge pull request #14 from iO-Academy/s4-t1-reveal-answer

Created reveal button functionality

ros-a committed:
added functionality next button and added design

SamBellDev committed:
Merge branch 's4-game-progress' into S4-T2-Next-Question

ros-a committed:
added line to submit button event

ros-a committed:
Merge branch 'S4-T2-Next-Question' of github.com:iO-Academy/2021-mongooses-guess-the-film into S4-T2-Next-Question

SamBellDev committed:
bug fix

ros-a committed:
changed the feedback

ros-a committed:
Merge branch 'S4-T2-Next-Question' of github.com:iO-Academy/2021-mongooses-guess-the-film into S4-T2-Next-Question
want to push

SamBellDev committed:
Merge pull request #15 from iO-Academy/S4-T2-Next-Question

added functionality next button and added design

ashleycoles commented on index.html:
```diff
@@ -5,62 +5,67 @@
+            <footer id="next_question" class="next_question">
+                <button id="next_question_button">Next</button>
+            </footer>
+        </section>
```
There's an unclosed div before this closing section tag

ashleycoles commented on index.js:
```diff
@@ -58,3 +70,8 @@ document.getElementById('start_button').addEventListener('click', (e) => {
+document.getElementById('reveal_button').addEventListener('click', (e) => {
+    document.getElementById('answer').style.display ='block'
+    document.getElementById('next_question_button').disabled = false
+    document.getElementById('submit_button').disabled = true
```
I suggest also changing the cursor when you disable the button

ashleycoles commented on index.js:
```diff
@@ -46,6 +50,14 @@ document.getElementById('submit_button').addEventListener('click', function () {
 
+document.getElementById('next_question_button').addEventListener('click', function () {
+    document.getElementById('reveal_button').disabled = false
+    document.getElementById('submit_button').disabled = false
+    get_new_question()
+    document.getElementById('next_question_button').disabled = true
+    document.getElementById('submit_button').disabled = false
```
Since you're selecting the submit button twice here I'd suggest assigning it to a variable to make the code more DRY

ashleycoles replied:
You could even take this a step further by putting any DOM queries (getElementByIds) that are repeated multiple times into variables in the global scope and then referencing those variables instead of reselecting them all the time. Again just to make things a bit more DRY - this one is optional though!

ashleycoles commented on index.js:
```diff
@@ -46,6 +50,14 @@ document.getElementById('submit_button').addEventListener('click', function () {
 })
 
+document.getElementById('next_question_button').addEventListener('click', function () {
+    document.getElementById('reveal_button').disabled = false
```
Might be worth clearing the answer box when you hit next? If you have something entered, then you reveal and hit next, the text is still there

ashleycoles commented on index.js:
```diff
@@ -1,3 +1,6 @@
+document.getElementById('end_screen').style.display = 'none'
+document.getElementById('game_screen').style.display = 'none'
```
Since these default to none, I would suggest setting it in the CSS rather than JS just because it's more efficient

ashleycoles reviewed (changes requested):
Some changes here for you!

As a general comment, some of the functions and event listeners are starting to get quite large, have a look for opportunities to split them down into smaller functions!

MrLukey committed:
Trainer Review Changes - 2

Added closing <div> tag for user_feedback element

MrLukey committed:
Trainer Review Changes

1 - cleared the guess_input when clicking next_question_button
2 - removed a duplicate line of code
3 - changed cursor for submit_button to default style when submit_button is disabled

MrLukey committed:
Sorted trainer review changes

MrLukey committed:
Merge branch 's4-game-progress' of github.com:iO-Academy/2021-mongooses-guess-the-film into s4-game-progress
Trainer review changes

MrLukey committed:
Changed cursor functionlaity to remove pointer style when button is disabled
